repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -364,29 +450,52 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
       val x509Data = keyInfoFactory.newX509Data(List(x509Certificate.getSubjectX500Principal.getName, x509Certificate).asJava)
       keyInfo = keyInfoFactory.newKeyInfo(Collections.singletonList(x509Data))
     } catch {
-      case e @ (_ : NoSuchMechanismException |
-                _ : ClassCastException |
-                _ : IllegalArgumentException |
-                _ : GeneralSecurityException |
-                _ : KeyStoreException |
-                _ : IOException) => throw new UpdateFailedException("Failed to load the signing credentials.", e)
+      case e@(_: NoSuchMechanismException |
+              _: ClassCastException |
+              _: IllegalArgumentException |
+              _: GeneralSecurityException |
+              _: KeyStoreException |
+              _: IOException) => throw new UpdateFailedException("Failed to load the signing credentials.", e)
     }
+
+    samlIdentityClient.using(
+      newConfiguration.getPolicyAcquisition.getKeystoneCredentials.getUri,
+      newConfiguration.getPolicyAcquisition.getPolicyEndpoint.getUri,
+      Option(newConfiguration.getPolicyAcquisition.getKeystoneCredentials.getConnectionPoolId),
+      Option(newConfiguration.getPolicyAcquisition.getPolicyEndpoint.getConnectionPoolId)
+    )
   }
 
   /**
     * Unsubscribe from the Atom Feed service.
     */
   override def doDestroy(): Unit = {
+    configurationService.unsubscribeFrom(SystemModelConfig, SystemModelConfigListener)
+
     if (feedId.nonEmpty) {
```

Comment:
Not your code, but this totally doesn't need to be an if. it can just be a foreach